Path:  Refactored PostUtilsParse.parse a lot.

LarryWoestman opened the pull request:
This is the next chunk of changes for the "refactored" postprocessor work.

This is part of the work for https://github.com/FreeCAD/FreeCAD/issues/6117.

Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [X]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

sliptonic commented:
I'd like to see more descriptive commit messages. There's only one commit in this PR and the description is fine for the first line.   Then there's effectively no body.  The commit history, as a whole, should be readable and almost tell a story about the evolution of the project.  So each message needs to carry its weight.

for reference: https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/

Similarly, I'd like to see a better description of the PR.  "This is the next chunk of ..."  Doesn't tell me what's going on in THIS chunk.  Anything that helps the reviewer understand the code/change is good.

for reference: https://levelup.gitconnected.com/how-to-write-the-perfect-pull-request-d044625ace98

LarryWoestman commented:
I will try to figure out better commit messages and descriptions for this PR.

LarryWoestman commented:
This is one in a series of PR's where I am taking code from at least 4 postprocessors partially written by different people, combining them into three files, then trying to make the code more consistent and simpler.  This particular PR doesn't add any functionality but it should make the code more maintainable.

This was originally checked in as one commit, but after looking at it, it probably should have been done as several commits, one for each type of refactoring that I did.  These were the important changes:

- Modified the create_comment function to look up the the comment symbol instead of passing it as a parameter.  This resulted in simpler (and shorter) create_comment calls.
- Added format_for_axis, format_for_feed, and format_for_spindle functions that returned a number formatted with the proper  precision.  This simplified the code and made it more obvious which type of number formatting is wanted.
- Refactored the "parse" routine to pull out the code that handled a group (a compound or a project) into the "parse_a_group" routine and renamed the remaining code in the "parse" routine into the "parse_a_path" routine which handles a simple path.
- Refactored the "parse" routing to pull out the code that added the parameters (sometimes called "words") to the outstring into a "add_parameters" routine.  This shortened the "parse" routine and reduced the number of levels of indentation needed.
- Simplified the "add_parameters" routine by waiting until the end of the main loop to output the parameter letter followed by the parameter number value instead of having the output scattered through the code.
